pandas/core/dtypes/common.py

```diff
@@ -599,10 +599,13 @@ def is_dtype_equal(source, target) -> bool:
         if not isinstance(source, str):
             # GH#38516 ensure we get the same behavior from
             #  is_dtype_equal(CDT, "category") and CDT == "category"
+            # GH-44276 ensures behaviour is consistent
+            # is_dtype_equal (dtype, "string[pyarrow]")
             try:
                 src = get_dtype(source)
-                if isinstance(src, ExtensionDtype):
-                    return src == target
+                if isinstance(src, (ExtensionDtype, np.dtype)):
+
+                    return type(src) == type(target)
             except (TypeError, AttributeError):
```

Comment:
I think you can probably catch the ImportError here (so it always returns False if the above raises an ImportError)

Reply:
Done